src/server/ws-routes/controllers/Friends.js

```diff
@@ -0,0 +1,121 @@
+const namespaceInfo = require('../../protocol/protocol.json').friends
+const EVENT_TYPES = namespaceInfo.eventTypes
+const logger = require('../../logger')
+const _ = require('lodash')
+class Friends {
+	constructor({
+		UserModel = require('../../models/user')
+	} = {}) {
+		this.UserModel = UserModel
+	}
+
+	[EVENT_TYPES.CONNECTION](socket, connections) {
+		const username = socket.request.user.username
+		connections.usersToConnectionsMap.set(username, socket)
+	}
+
+	[EVENT_TYPES.FRIENDS_LIST](data, socket) {
+		const {username} = socket.request.user
+		return this.UserModel.findOne({username}).exec()
+			.then(user => socket.emit(EVENT_TYPES.FRIENDS_LIST, user.friends))
+			.catch(err => logger.error(err))
+	}
+
+	[EVENT_TYPES.INVITE](data, socket, connections) {
+		const invitatedUsername = data.username
+		const invitatingUsername = socket.request.user.username
+		const emitPayload = {username: invitatingUsername}
+		return this.addNotification(invitatingUsername, invitatedUsername, EVENT_TYPES.INVITE)
+			.then(() => this.sendMessageToSepcificUser(socket, connections, invitatedUsername, EVENT_TYPES.INVITE, emitPayload))
+			.catch(err => logger.error(err))
```

Comment:
Will admin that notices errors in the app will understand what happened? Shouldn't we add some additional information, like what happened before error?

Reply:
ok